msupply-foundation/open-msupply
# 5806 change button orange text on white to black text

Fixes #5806

# 👩🏻‍💻 What does this PR do?


Changes the orange text on buttons to black, needed for GEDSI improvements

Added a separate style for the text color and icon color, so that the icon remains orange.
On buttons that have 'hover' the hover remains the same as it was (either orange background/white text & icon or white background/orange text & icon)




Some examples:

https://github.com/user-attachments/assets/7ac2d1f1-517d-4f0d-9313-a5d5d34399e0
Hover states of Primary buttons

![Screenshot 2024-12-23 at 2 17 42 PM](https://github.com/user-attachments/assets/17413791-1926-4535-afac-c1698f880b74)
Flat buttons in Actions footer

![Screenshot 2024-12-23 at 2 17 28 PM](https://github.com/user-attachments/assets/37a6e3da-148c-4e7a-a2a7-74866cc98cf0)
(top) Base button with Icon
(right) Flat button with no icon

![Screenshot 2024-12-23 at 2 56 41 PM](https://github.com/user-attachments/assets/fb494cb3-9bd2-46eb-b0bd-877649f20af8)
Sidebar Actions flat buttons

Split buttons will also have the new styling of the base button if set to Primary color, but are currently used with the 'secondary' blue color

Loading buttons, dialog buttons are sometimes set to Primary but with orange background/white text - no change

## 💌 Any notes for the reviewer?


##### Flat Buttons:
The FlatButton component did not have anything set up to be able to set colors - it was 'default' color which did not enable changing the text color the same way as the Base Button. For consistency I have added the ability to pass a color selection through as props, but if none passed it will default to Primary

- in Detail View -> Print -> ReportSelector: these have no icon so have set color to 'Primary' to show black text only. Being a flat button, they have no hover state. Do they look enough like a clickable button for the user? 







# 🧪 Testing


These buttons are throughout the app. Some places that have multiple examples are:

Outbound Shipments -> Detail View:
- Top right corner are new style base buttons with icon, hover for orange button with white text & icon
- Select Print: new style flat buttons without icon
- Select More: new style flat buttons with icon (actions)
- Select a shipment: footer has new style flat buttons with icon
- Click delete - see that these buttons have not changed - are blue and white

Settings:
- see the buttons remain orange with white text, and reverse to white button with orange text & icon on hover


# 📃 Documentation

  1.
  2.
